### PR TITLE
roachpb: deprecate TxnCoordMeta.RefreshValid, replace with RefreshInvalid

### PR DESCRIPTION
**Pull request by nvb**
This change replaced `TxnCoordMeta.RefreshValid` with a new
`TxnCoordMeta.RefreshInvalid` field. The change may seem pointless,
but its actually pretty important because it gives TxnCoordMeta (and
by extension, txnSpanRefresher) a useful zero value [1]. A later commit
in this PR will construct these `TxnCoordMeta` objects more often, so
this is important.

Deprecating the field and inverting it safely is a little tricky. It
requires a new cluster version. For now, all 2.1 nodes will continue
to set `DeprecatedRefreshValid` flags. 2.1 nodes who observe a cluster
version beneath the new `VersionTxnCoordMetaInvalidField` will use
`DeprecatedRefreshValid` to distinguish between an old node indicating
that refresh spans are invalid and a new node indicating that refresh
spans are valid. This is possible because old nodes will create a
contradiction in this scenario where both fields are set to `false`.

We can remove the deprecated field entirely in 2.2 binaries. This is
possible because these binaries will never talk to nodes who think
the cluster version is less than `VersionTxnCoordMetoaInvalidField`.

[1] https://dave.cheney.net/2013/01/19/what-is-the-zero-value-and-why-is-it-useful

Release note: None

### CODE REVIEW

**Comment by nvanbenschoten** [comment]
@bdarnell mind if I add you as a reviewer since we just discussed this and you probably have it paged in?

**Review by RaduBerinde — APPROVED** [review]
I started reviewing this in 27066 before realizing it's a separate PR. Moving over the small comments I had.

:lgtm:

As long as the deprecated flag is in there, the zero value for `TxnCoordMeta` is technically not valid. Maybe mention in the commit how that works (we only care about the deprecated flag at serialization/deserialization time).

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/27420#-:-LHDPs1jEjVbUQ9taNmv:b1c5hne)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

___
*[pkg/roachpb/data.proto, line 483 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/27420#-LHDQ5Th0WFiopHaZeJR:-LHDQ5Th0WFiopHaZeJS:b-ja13z2) ([raw file](https://github.com/cockroachdb/cockroach/blob/34a5c0df7dec70b1f8a35b59de51871f21ef29b5/pkg/roachpb/data.proto#L483)):*
> ```protobuf
>   bool refresh_invalid = 7;
>   // deprecated_refresh_valid is the inverse of refresh_invalid. It was
>   // deprecated because it resulted in a useless zero value.
> ```

[nit] It's not clear that `zero value` refers to the value of the struct. Maybe rephrase to `it was deprecated in favor of refresh_invalid in order to give the struct a useful zero value`.

**Comment by RaduBerinde** [comment]
___
*[pkg/kv/txn_interceptor_span_refresher.go, line 358 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/27420#-LHDrfLo5sN439g4N9rw:-LHDrfLp3Xf58gWVRfh7:b79zfk4) ([raw file](https://github.com/cockroachdb/cockroach/blob/34a5c0df7dec70b1f8a35b59de51871f21ef29b5/pkg/kv/txn_interceptor_span_refresher.go#L358)):*
> ```Go
> 		// DeprecatedRefreshValid and RefreshInvalid are being set correctly.
> 		// TODO(nvanbenschoten): Can be removed in 2.2.
> 		if meta.RefreshInvalid == meta.DeprecatedRefreshValid {
> ```

It is not possible that both are true, correct? I'd make the condition `if !meta.RefreshInvalid && !meta.DeprecatedRefreshValid` for clarity (I initially thought that there are two possibilities captured by the condition).

**Review by bdarnell — APPROVED** [review]
> Reviewed 10 of 10 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/27420)* status: :shipit: complete! 1 of 0 LGTMs obtained

**Review by nvb — COMMENTED** [review]
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/27420#-:-LHEoJ9n6hxSdgWCFSlp:b-814g8k)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

___
*[pkg/kv/txn_interceptor_span_refresher.go, line 358 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/27420#-LHDrfLo5sN439g4N9rw:-LHEnpkgC-Uz7KUqg9TH:bpjpkxk) ([raw file](https://github.com/cockroachdb/cockroach/blob/34a5c0df7dec70b1f8a35b59de51871f21ef29b5/pkg/kv/txn_interceptor_span_refresher.go#L358)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

It is not possible that both are true, correct? I'd make the condition `if !meta.RefreshInvalid && !meta.DeprecatedRefreshValid` for clarity (I initially thought that there are two possibilities captured by the condition).
</blockquote></details>

Exactly, sorry for the confusion. Done.
___
*[pkg/roachpb/data.proto, line 483 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/27420#-LHDQ5Th0WFiopHaZeJR:-LHEntSM3c0yFMTs5IJf:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/34a5c0df7dec70b1f8a35b59de51871f21ef29b5/pkg/roachpb/data.proto#L483)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

[nit] It's not clear that `zero value` refers to the value of the struct. Maybe rephrase to `it was deprecated in favor of refresh_invalid in order to give the struct a useful zero value`.
</blockquote></details>

Done.

**Review by RaduBerinde — APPROVED** [review]
> Reviewed 7 of 10 files at r1, 3 of 3 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/27420)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)
